source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -623,10 +627,32 @@ def call_kraken(self, request_type, request, instance, krakens_call):
 
         worker = AsyncWorker(instance, krakens_call, request)
 
+        resp = []
+
+        # Now we compute the direct path with all requested departure mode
+        # their time will be used to initialized our PT calls
+        futures = worker.get_direct_path_futures(g.fallback_direct_path,
+                                                 g.requested_origin,
+                                                 g.requested_destination,
+                                                 request['datetime'],
+                                                 request['clockwise'],
+                                                 False,
+                                                 {mode for mode, _ in krakens_call})
+        for future in gevent.iwait(futures):
+            resp_key, resp_direct_path = future.get()
+            g.fallback_direct_path[resp_key] = resp_direct_path
+            if resp_direct_path.journeys:
+                resp_direct_path.journeys[0].internal_id = str(generate_id())
+                resp.append(resp_direct_path)
+
         if request.get('max_duration', 0):
+            direct_path_duration_by_mode = make_direct_path_duration_by_mode(g.fallback_direct_path)
+
             # Get all stop_points around the requested origin within a crowfly range
```

Comment:
Sorry for bothering, i think we can actually call `get_max_fallback_duration` immediately after the computation of the "general" direct_path, instead of calling it repeatedly in `get_crowfly_futures` and `get_routing_matrix_futures` (we can do it later, i'm ok with that )